src/base/pegasus_value_schema.h

```diff
@@ -238,13 +238,24 @@ class pegasus_value_generator
 
 enum data_version
 {
+    VERSION_0 = 0,
+    VERSION_COUNT,
+    VERSION_MAX = VERSION_0,
     /// TBD(zlw)
 };
 
 struct value_params
 {
+    value_params(std::string &buf, std::vector<rocksdb::Slice> &slices)
+        : write_buf(buf), write_slices(slices)
+    {
+    }
+
     std::map<value_field_type, std::unique_ptr<value_field>> fields;
```

Comment:
Maybe we can replace it with `std::array" to get more efficiency by both time complexity(O(logN) vs O(1)) and [cache-friendly ](https://stackoverflow.com/questions/16699247/what-is-a-cache-friendly-code).

Reply:
```
std::map<value_field_type, std::unique_ptr<value_field>> fields;
```
=>
```
std::array<std::unique_ptr<value_field>> fields;
// fields[value_field_type] = dsn::make_unique<time_tag_field>(time_tag);
```

Reply:
done